docs/source/development-environment.rst

```diff
@@ -0,0 +1,141 @@
+Zipline Development Guidelines
```

Comment:
It feels like we probably want an opening paragraph explaining what this page is about and who it's for.  Specifically, we should emphasize that the resources here are for users who want to contribute to Zipline or who want to install from source to make local changes to their copy of Zipline.